presto-main/src/main/java/com/facebook/presto/cost/StatisticRange.java

```diff
@@ -36,8 +38,8 @@ public StatisticRange(double low, double high, double distinctValues)
     {
         checkState(low <= high || (isNaN(low) && isNaN(high)), "Low must be smaller or equal to high or range must be empty (NaN, NaN)");
         checkState(distinctValues >= 0 || isNaN(distinctValues), "Distinct values count cannot be negative");
-        this.low = low;
-        this.high = high;
+        this.low = isNaN(low) ? NEGATIVE_INFINITY : low;
```

Comment:
I do not see a reason when low and high could be NaN, so I would add here a check that low and high are not Nan and fix the other places.